lib/TPP/Transforms/TileConfig.cpp

```diff
@@ -0,0 +1,154 @@
+      return failure();
+    auto flags =
+        dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp()).getFlags();
+    for (auto flagItr : flags) {
+      if (flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_RESET_TILECONFIG) ||
+          flagItr == xsmm::GemmFlagsAttr::get(
+                         rewriter.getContext(),
+                         mlir::xsmm::GemmFlags::NO_SETUP_TILECONFIG)) {
+        return failure();
+      }
+    }
+
+    SmallVector<Attribute> attributesSetup;
+    attributesSetup.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_RESET_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesSetup, rewriter, op);
+    auto tileConfigSetup = rewriter.create<xsmm::TileConfigDispatchOp>(
+        op.getLoc(), rewriter.getI64Type(),
+        DenseI64ArrayAttr::get(
+            rewriter.getContext(),
+            dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp())
+                .getInputs()),
+        rewriter.getArrayAttr(attributesSetup),
+        xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()));
+
+    SmallVector<Attribute> attributesReset;
+    attributesReset.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_SETUP_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesReset, rewriter, op);
+    auto tileConfigReset = rewriter.create<xsmm::TileConfigDispatchOp>(
+        op.getLoc(), rewriter.getI64Type(),
+        DenseI64ArrayAttr::get(
+            rewriter.getContext(),
+            dyn_cast<DispatchOpTy>(op.getOperand(0).getDefiningOp())
+                .getInputs()),
+        rewriter.getArrayAttr(attributesReset),
+        xsmm::utils::getDataType(rewriter, op.getOperand(1).getType()));
+
+    SmallVector<Attribute> attributesBrgemm;
+    attributesBrgemm.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_RESET_TILECONFIG));
+    attributesBrgemm.push_back(xsmm::GemmFlagsAttr::get(
+        rewriter.getContext(), xsmm::GemmFlags::NO_SETUP_TILECONFIG));
+    appendBrgemmFlags<InvokeOpTy, DispatchOpTy>(attributesBrgemm, rewriter, op);
+
+    auto dispatch = dyn_cast<DispatchOpTy>(
+        rewriter.clone(*op.getOperand(0).getDefiningOp()));
+    dispatch.setFlagsAttr(rewriter.getArrayAttr(attributesBrgemm));
+
+    auto alloca = rewriter.create<memref::AllocaOp>(
+        op.getLoc(), MemRefType::get({64}, rewriter.getI8Type()));
+    
+    ValueRange tileConfigInputs{alloca};
+    rewriter.create<mlir::xsmm::TileConfigOp>(
+        op.getLoc(), tileConfigSetup, tileConfigInputs);
+    
+    SmallVector<Value> invokeOperands;
+    invokeOperands.push_back(dispatch);
+    auto opItr = op->getOperands().begin();
+    std::advance(opItr, 1);
+    invokeOperands.append(opItr, op->getOperands().end());
```

Comment:
Do you need to re-create the original GEMM/BRGEMM op? They're independent of the tile config, you can just use the rewriter's ability to change the insertion point (before/after) and add each tile config invoke.

Reply:
I could just modify the existing op's dispatch, yes. I'll update the code.

Reply:
> They're independent of the tile config, you can just use the rewriter's ability to change the insertion point (before/after) and add each tile config invoke.

No! The tileconfig/tilerreset need to be around the BRGEMM. Hence, I think fiddling with inserting points is brittle, just let's rewrite and replace.